src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -615,30 +615,50 @@ FixedwingPositionControl::control_position(const Vector2f &curr_pos, const Vecto
 		_tecs.set_speed_weight(_param_fw_t_spdweight.get());
 		_tecs.set_time_const_throt(_param_fw_t_thro_const.get());
 
-		/* current waypoint (the one currently heading for) */
-		Vector2f curr_wp((float)pos_sp_curr.lat, (float)pos_sp_curr.lon);
+		Vector2f curr_wp{0.0f, 0.0f};
+		Vector2f prev_wp{0.0f, 0.0f};
 
-		/* Initialize attitude controller integrator reset flags to 0 */
-		_att_sp.roll_reset_integral = false;
-		_att_sp.pitch_reset_integral = false;
-		_att_sp.yaw_reset_integral = false;
+		if (_vehicle_status.in_transition_to_fw) {
 
-		/* previous waypoint */
-		Vector2f prev_wp{0.0f, 0.0f};
+			if (!PX4_ISFINITE(_transition_waypoint(0))) {
+				double lat_transition, lon_transition;
+				waypoint_from_heading_and_distance(_current_latitude, _current_longitude, _yaw, 1000.0f, &lat_transition,
```

Comment:
A small comment where this 1000 is coming from would be useful here. Maybe best define a separate variable for it, e.g. const float transition_destination_dist = 1000.0f.

Reply:
@sfuhrer yes, I can do that